Test/VimCoreTest/VisualModeIntegrationTest.cs

```diff
@@ -1204,8 +1204,20 @@ public void DontRepeatDelete()
                     Assert.Equal("og", _textView.GetLine(0).GetText());
                     Assert.Equal("cat", _textView.GetLine(1).GetText());
                 }
-            }
 
+                /// <summary>
+                /// When the block selection is in column zero then empty lines need to 
+                /// get the block insert applied as well
+                /// Issue 2342
+                /// </summary>
+                [WpfFact]
+                public void EmptyLineBlockAtStartOfLine()
+                {
+                    Create("dog", "", "tree");
+                    _vimBuffer.ProcessNotation(@"<C-q>jjI#<Esc>");
+                    Assert.Equal(new[] { "#dog", "#", "#tree" }, _textBuffer.GetLines());
+                }
+            }

```

Comment:
A similar test from vim column 2 with the buffer:

```
dog
x
tree
```

should result in:

```
d#og
x#
t#ree
```

but with the current code change, the middle line doesn't get an insertion:

```
d#og
x
t#ree
```

Reply:
It would also be good to have to a test for when we *correctly* don't perform an insertion, that is when the block is truly *past* the end-of-line. An example is the buffer:

```
dog

tree
```

from vim column 2.

Reply:
> but with the current code change, the middle line doesn't get an insertion:

Good catch. Let me add that. 

> It would also be good to have to a test for when we correctly don't perform an insertion, that is when the block is truly past the end-of-line

That test is already present: `BlockInsertTest.MiscTest.EmptyColumn`